.github/workflows/sync-ruff-config.yml

```diff
@@ -0,0 +1,50 @@
+name: Sync Ruff Configuration
+
+on:
+  schedule:
+    - cron: "0 0 * * 0" # Runs every week
+  workflow_dispatch:
+
+jobs:
+  sync-file:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout this repository
+        uses: actions/checkout@v4
+
+      - name: Update or create sync branch
+        run: |
+          git fetch origin master
+          if git ls-remote --exit-code --heads origin sync-ruff-config; then
+            git checkout sync-ruff-config
+            git reset --hard origin/master
+          else
+            git checkout -b sync-ruff-config
+          fi
+
+      - name: Fetch ruff.toml from NCTR's Python Development Guide
+        run: |
+          curl -o ruff.toml https://raw.githubusercontent.com/NationalCentreTruthReconciliation/Python-Development-Guide/main/ruff.toml
```

Comment:
These GH actions are a bit abstract to me, but I think the directory this command runs in is the same as the root of the repository, right?

If that's the case, I have another issue I flagged below.